src/test/java/com/epam/tc/hw4/utils/AssertStep.java

```diff
@@ -0,0 +1,122 @@
+package com.epam.tc.hw4.utils;
+
+import static com.epam.tc.hw4.tests.BaseTest.driver;
+import static org.assertj.core.api.Assertions.assertThat;
+import static pages.Header.DIFF_ELEMENTS_TITLE;
+import static pages.Header.HOME_PAGE_TITLE;
+import static pages.Header.headerNamesExpected;
+import static pages.Header.userNameLogged;
+import static pages.IndexPage.expectedText;
+import static pages.LeftSection.names;
+
+import io.qameta.allure.Step;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.testng.asserts.SoftAssert;
+import pages.DiffElementsPage;
+import pages.Header;
+import pages.IndexPage;
+import pages.LeftSection;
+
+public class AssertStep extends PageObject {
+    SoftAssert softAssert = new SoftAssert();
+    private static final String iFrameElement = "frame";
+
+    public AssertStep(WebDriver driver) {
+        super(driver);
+    }
+
+    @Step("Assert title of the page")
+    public void assertTitle() {
+        assertThat(driver.getTitle()).as("Wrong Page title").isEqualTo(HOME_PAGE_TITLE);
+    }
+
+    @Step("Assert login")
+    public void assertLogin() {
+        assertThat(userNameLogged).as("Wrong username is logged").isEqualTo("ROMAN IOVLEV");
+        softAssert.assertAll();
+    }
+
+    @Step("Assert login (failed)")
+    public void assertLoginFailed() {
+        assertThat(userNameLogged).as("Wrong username is logged").isEqualTo("ROMAN IOVLEVV");
+        softAssert.assertAll();
+    }
+
+    @Step("Assert 4 items in the header section with proper text")
+    public void assertHeader() {
+        header = new Header(driver);
+        assertThat(header.getItemsTextInHeader()).as("Wrong headers are displayed").isEqualTo(headerNamesExpected);
+    }
+
+    @Step("Assert 4 images on the index page")
+    public void assertImages() {
+        indexPage = new IndexPage(driver);
+        for (WebElement i : indexPage.getImagesDisplayed()) {
+            softAssert.assertTrue(i.isDisplayed());
+        }
```

Comment:
assertAll()

Reply:
After all assertions in a method, put it after for loop (to collect all verifications and finalize them at the end of a method)